backend/fms_core/migrations/0064_v4_11_0.py

```diff
@@ -58,6 +60,40 @@ def initialize_default_reference_genome(apps, schema_editor):
             taxon_obj.save()
             reversion.add_to_revision(taxon_obj)
 
+def initialize_validated_by_and_released_by(apps, schema_editor):
+    Readset = apps.get_model("fms_core", "Readset")
+    Version = apps.get_model("reversion", "Version")
+
+    readsets_validated_ids = list(Readset.objects.exclude(validation_status_timestamp__isnull=True).values_list("id", flat=True) or [])
+    readsets_released_ids = list(Readset.objects.exclude(release_status_timestamp__isnull=True).values_list("id", flat=True) or [])
+    readsets_to_update_ids = set(readsets_validated_ids + readsets_released_ids)
+
+    with reversion.create_revision(manage_manually=True):
+        admin_user = User.objects.get(username=ADMIN_USERNAME)
+
+        reversion.set_comment("Initialize the validated_by and released_by for readset validated and released before the fields were added.")
+        reversion.set_user(admin_user)
+
+        for readset_id in readsets_to_update_ids:
+            readset_obj = Readset.objects.get(id=readset_id)
+            validated_by_id = None
+            released_by_id = None
+            for readset_validated_version in Version.objects.filter(content_type__model="readset", object_id=readset_id).order_by('-id'):
+                data = json.loads(readset_validated_version.serialized_data)
+                validation_status_timestamp = data[0]["fields"].get("validation_status_timestamp", None)
+                release_status_timestamp = data[0]["fields"].get("release_status_timestamp", None)
+                if validated_by_id is None and validation_status_timestamp is not None:
+                    validated_by_id = data[0]["fields"]["updated_by"]
+                if released_by_id is None and release_status_timestamp is not None:
+                    released_by_id = data[0]["fields"]["updated_by"]
```

Comment:
Maybe you can go in the opposite direction (`order_by('id')`) so you can exit this loop sooner?

Reply:
object id are given in order of creation ... latest ones are most recent ... i want the latest one not the earliest. It is a bit lazy since it does not validate if both information are required during the loop but i don't expect millions of versions...